### ISSUE TITLE
Get hold of the instantiated instance of the component

### ISSUE THREAD

**Issue by abdulqadir93** [makinacorpus/angular-traversal#3]
Since the components are instantiated by the library, I couldn't get hold of the instance of the active  component. I had to make use of a service to communicate between the components. Getting a hold of the instance should make it easy to pass inputs etc.

**Comment by ebrehault** [comment]
Do you mean you would like to get access to the parent component (the one containing the `<traverser-outlet>`) from the View component?

**Comment by abdulqadir93** [comment]
No. I probably should've added a code snippet in the description. Consider a hypothetical component below:

```
@Component({
  selector: "parent",
  template: `
    <header (onRefresh)="refresh()"></header>
    <div id="content">
      <sidebar [item]="item"></sidebar>
      <traverser-outlet></traverser-outlet>
    </div>
  `
})
export class Parent {
    item: Item;
    activeComponent: IRefreshComponent;

    ngOnInit() {
        traverser.addView(...);
        ....
        traverser.target.subscribe(target => {
            this.item = target.context;
            this.activeComponent = target.component; //not the actual instance of component!!
        });
    }
    refresh() {
        if (this.activeComponent) {
            // if activeComponent was an actual instance, the line below would've worked
            // this.activeComponent.refresh();
            // the workaround is to communicate via service which although works but is a lot of work.
        }
    }
}
```

Now if you see, the header component emits an event, which needs to be communicated to the component inside the traverser outlet, so that it can refresh the results (or whatever is inside the component). But `target.component` doesn't hold the instance of the component. 

The target-outlet does have a `viewInstance` variable but it is private. Also can't get a hold of the directive since there's no exportAs field in the directive's metadata and thus template variable `<traverser-outlet #traverserOutlet="TraverserOutlet"></traverser-outlet>` won't work.

Please let me know if I'm not explaining it right

**Comment by ebrehault** [comment]
OK I see.
In such a case, I would definitely prefer to communicate through a service, but yes, I agree that could be useful. Let's do it.
